rbd.c

```diff
@@ -836,32 +851,39 @@ static int tcmu_rbd_write(struct tcmu_device *dev, struct tcmulib_cmd *cmd,
 	aio_cb->tcmulib_cmd = cmd;
 	aio_cb->read = false;
 
-	aio_cb->bounce_buffer = malloc(length);
-	if (!aio_cb->bounce_buffer) {
-		tcmu_dev_err(dev, "Failed to allocate bounce buffer.\n");
+	ret = rbd_aio_create_completion
+		(aio_cb, (rbd_callback_t) rbd_finish_aio_generic, &completion);
+	if (ret < 0) {
 		goto out_free_aio_cb;
 	}
 
-	tcmu_memcpy_from_iovec(aio_cb->bounce_buffer, length, iov, iov_cnt);
-
-	ret = rbd_aio_create_completion
-		(aio_cb, (rbd_callback_t) rbd_finish_aio_generic, &completion);
+#ifdef RBD_IOVEC_SUPPORT
```

Comment:
I know it ends up being more code, but we had been trying to follow the kernel coding style as much as we can, so when we flip back and forth between runner and tcmu we do not have to adjust our styles.

I just merged how we handle ifdefs in this patch (instead of adding them in the function we do the wrapper functions):

https://www.spinics.net/lists/target-devel/msg15294.html

and your patch into this new combined PR:

https://github.com/open-iscsi/tcmu-runner/pull/331